kolibri/core/auth/management/commands/importusers.py

```diff
@@ -119,9 +119,10 @@ def create_user(user, default_facility=None):
 
     except FacilityUser.DoesNotExist:
         password = user.get("password", DEFAULT_PASSWORD) or DEFAULT_PASSWORD
+        full_name = user.get("full_name", "") or username
```

Comment:
shouldn't this just be

```python
full_name = user.get("full_name", username)
```

I don't really understand the use of the `or` in this line or the `password` line above. If the point is to avoid empty strings, I think it's much more explicit to use a ternary statement:

```python
full_name = user.get("full_name") if user.get("full_name") else username
```

(Statements with `or` in them look like they return booleans to me, not a way of choosing default values)

Reply:
The point is to always set a `full_name` regardless of whether it has been passed in as blank, similarly for password. If it has been passed in as blank, it still sets the default password.

Doing `user.get("full_name")` will throw an `AttributeError` if `user` does not have this field, whereas having the default value will prevent this.

As such, to achieve the goal of having a non-empty `full_name` and `password` regardless of input, the code that is here is best - the other alternative would be:

`full_name = user.get("full_name", "") if user.get("full_name", "") else username`

Reply:
This is a nice summary of how boolean operators map to conditional expressions: https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not

Reply:
Maybe using `or` as a way of choosing defaults is more idomatic than I thought? Not sure.

While we're nerding out, I think this one is actually my favorite, though certainly the most verbose:

```python
full_name = user.get("full_name", None)
if not full_name:
    full_name = username
```

Reply:
My sense is that `or` is quite Pythonic, at the very least.

Reply:
> Doing user.get("full_name") will throw an AttributeError if user does not have this field, whereas having the default value will prevent this.

Reply:
at least [some people on the internet](https://stackoverflow.com/a/13710667) consider `value or default` to be an anti-pattern, but who knows...

my current favorite given all of the above:

```python
full_name = user.get("full_name") if user.get("full_name") else username
```

Reply:
My personal dislike for the conditional expression/ternary is the repeated evaluation - although given that it is a dict getter, this is not particularly important in this instance.